p3analysis/_utils.py

```diff
@@ -19,14 +19,15 @@ def _require_columns(df, columns):
             raise ValueError(msg % (column, str(columns)))
 
 
-def _require_numeric(df, columns):
+def _cast_to_numeric(df, columns):
     """
-    Check that the named columns are numeric.
+    Check that the named columns are numeric, and cast them.
     """
-
+    result = df.copy(deep=True)
     for column in columns:
         try:
-            pd.to_numeric(df[column])
+            result[column] = pd.to_numeric(df[column])
         except Exception:
```

Comment:
My only gripe with this is that a `KeyError` (column doesn't exist) will trigger this as well

Reply:
Fixed in 8367638463155951194dcf780f609e766651e36c.

Reply:
I had to rebase to resolve conflicts introduced by #70.